### ISSUE TITLE
Dataclass conversion error (regression introduced on 0.7.6)

### ISSUE THREAD

**Issue by 0rtis** [ApeWorX/ape#2008]
### Environment information

- OS:  linux
- Python Version: 3.11.9
- `ape` and plugin versions:

```sh
$ ape --version
# 0.7.15

$ ape plugins list
# Installed Plugins
  solidity    0.7.1
```

- Contents of your `ape-config.yaml`:

```sh
$ cat ape-config.yaml
# name: regression
default_ecosystem: ethereum
test:
  mnemonic: "test test test test test test test test test test test junk"
  number_of_accounts: 10
compiler:
  ignore_files:
    - "*.md"
    - "*.adoc"
plugins:
  - name: solidity
solidity:
  version: 0.8.24

```

### What went wrong?

A regression was introduced on data class conversion starting version 0.7.6 of eth-ape. Retrieving a struct cause an error `TypeError: non-default argument 'datas' follows default argument`.

#### Reproducible example
The following example will cause an error when running `ape test` with any version >= 0.7.6
Version 0.7.5 is unaffected
##### Solidity file
```
//SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;


contract MultiSig {

   struct MultiSigCall {
        uint id;
        address [] targets;
        uint [] values;
        bytes [] datas;
        address [] approvals;
        bool executed;
    }

    mapping(uint => MultiSigCall) multiSigCalls;

    uint public nextProposalId = 1;


    function initMultiSigCall(MultiSigCall memory proposal) public returns(uint) {
        proposal.id = nextProposalId;
        multiSigCalls[nextProposalId++] = proposal;
        return proposal.id;
    }

    function getMultiSigCall(uint proposalId) public view returns(MultiSigCall memory) {
        return multiSigCalls[proposalId];
    }

    receive() external payable {}

    fallback() external payable {}

}

```

##### Test file

```
def test_multisig(accounts, project):

    eoa = accounts[0]
    multisig = eoa.deploy(project.MultiSig)


    original_multisig_call = (0, ['0xf091867EC603A6628eD83D274E835539D82e9cc8'], [0], ['0xa9059cbb000000000000000000000000a5c0067121b222d1303b00bdd750424874e8be6f0000000000000000000000000000000000000000000000000dfdf13250603400'], ['0x3524727d39a6344C44a33Ea99862c8791a703D25'], False)

    multisig.initMultiSigCall(original_multisig_call, sender=eoa)
    multisig_call = multisig.getMultiSigCall(1) # this fails with any version >= 0.7.6
```

**Comment by antazoey** [comment]
Trying to figure out what EVM version to use as well what provider.

* Tried running on `::test` but getting `Invalid opcode 0x5e @ 2297`
* Tried running on `::foundry` but got `EVM error NotActivated`
** Ran `foundryup` and retried: **was able to reproduce!**

**Comment by antazoey** [comment]
figured out it is related to `values` in the struct conflicting with` values` method.

**Comment by 0rtis** [comment]
@antazoey Thank you ! I suggest to ask for the test provider in the bug report template. It completely skipped my mind.

**Comment by fubuloubu** [comment]
@antazoey is this closes by #2009?
